library/ssl_tls13_generic.c

```diff
@@ -494,56 +494,53 @@ void mbedtls_ssl_set_outbound_transform( mbedtls_ssl_context *ssl,
 }
 #endif /* !MBEDTLS_SSL_USE_MPS */
 
-    /*
-     * The mbedtls_ssl_create_verify_structure() creates the verify structure.
-     * As input, it requires the transcript hash.
-     *
-     * The structure is computed per TLS 1.3 specification as:
-     *   - 64 bytes of octet 32,
-     *   - 33 bytes for the context string
-     *        (which is either "TLS 1.3, client CertificateVerify"
-     *         or "TLS 1.3, server CertificateVerify"),
-     *   - 1 byte for the octet 0x0, which servers as a separator,
-     *   - 32 or 48 bytes for the Transcript-Hash(Handshake Context, Certificate)
-     *     (depending on the size of the transcript_hash)
-     *
-     * This results in a total size of
-     * - 130 bytes for a SHA256-based transcript hash, or
-     *   (64 + 33 + 1 + 32 bytes)
-     * - 146 bytes for a SHA384-based transcript hash.
-     *   (64 + 33 + 1 + 48 bytes)
-     *
-     * The caller has to ensure that the buffer has this size.
-     */
-static void mbedtls_ssl_create_verify_structure(
-                                        unsigned char *transcript_hash,
+/*
+ * The ssl_create_verify_structure() creates the verify structure.
+ * As input, it requires the transcript hash.
+ *
+ * The structure is computed per TLS 1.3 specification as:
+ *   - 64 bytes of octet 32,
+ *   - 33 bytes for the context string
+ *        (which is either "TLS 1.3, client CertificateVerify"
+ *         or "TLS 1.3, server CertificateVerify"),
+ *   - 1 byte for the octet 0x0, which servers as a separator,
+ *   - 32 or 48 bytes for the Transcript-Hash(Handshake Context, Certificate)
+ *     (depending on the size of the transcript_hash)
+ *
+ * This results in a total size of
```

Comment:
This calculation should be moved to the definition of `MBEDTLS_SSL_VERIFY_STRUCT_MAX_SIZE`, which should be referred here.